tests/tests_fabric/utilities/test_seed.py

```diff
@@ -85,3 +86,12 @@ def test_backward_compatibility_rng_states_dict():
     assert "torch.cuda" in states
     states.pop("torch.cuda")
     _set_rng_states(states)
+
+
+@mock.patch("lightning.fabric.utilities.seed.torch.cuda.is_available", Mock(return_value=False))
+@mock.patch("lightning.fabric.utilities.seed.torch.cuda.get_rng_state_all")
+def test_collect_rng_states_if_cuda_init_fails(get_rng_state_all_mock):
+    """Test that the `torch.cuda` rng states are only requested if CUDA is available."""
+    get_rng_state_all_mock.side_effect = RuntimeError("The NVIDIA driver on your system is too old")
+    states = _collect_rng_states()
+    assert states["torch.cuda"] == []
```

Comment:
```suggestion
    assert "torch.cuda" not in states
```

If the above is accepted